docs/languages/csharp/expression-trees-building.md

```diff
@@ -0,0 +1,218 @@
+build the tree from the leaves up to the root. The APIs you'll use to
+build expression trees reflect this fact: The methods you'll use to
+build a node take all its children as arguments. Let's walk through
+a few examples to show you the techniques.
+
+## Creating Nodes
+
+Let's start relatively simply again. We'll use the addition
+expression I've been working with throughout these sections:
+
+```cs
+Expression<Func<int>> sum = () => 1 + 2;
+```
+
+To construct that expression tree, you must construct the leaf nodes.
+The leaf nodes are constants, so you can use the `Expression.Constant`
+method to create the nodes:
+
+```cs
+var one = Expression.Constant(1, typeof(int));
+var two = Expression.Constant(2, typeof(int));
+```
+
+Next, you'll build the addition expression:
+
+```cs
+var addition = Expression.Add(one, two);
+```
+
+Once you've got the addition expression, you can create the lambda
+expression:
+
+```cs
+var lamdba = Expression.Lambda(addition);
+```
+
+This is a very simple LambdaExpression, because it contains no arguments.
+Later in this section, you'll see how to map arguments to parameters
+and build more complicated expressions.
+
+For expressions that are as simple as this one, you may combine all the
+calls into a single statement:
+
+```cs
+var lambda = Expression.Lambda(
+    Expression.Add(
+        Expression.Constant(1, typeof(int)),
+        Expression.Constant(2, typeof(int))
+    )
+);
+```
+
+## Building a Tree
+
+That's the basics of building an expression tree in memory. More
+complex trees generally mean more node types, and more nodes in the
+tree. Let's run through one more example and show two more node types
+that you will typically build when you create expression trees:
+the argument nodes, and method call nodes.
+
+Let's build an expression tree to create this expression:
+
+```cs
+Expression<Func<double, double, double>> distanceCalc =
+    (x, y) => Math.Sqrt(x * x + y * y);
+```
+ 
+You'll start by creating parameter expressions for `x` and `y`:
+
+```cs
+var xParameter = Expression.Parameter(typeof(double), "x");
+var yParameter = Expression.Parameter(typeof(double), "y");
+```
+
+Creating the multiplication and addition expressions follows the pattern
+you've already seen:
+
+```cs
+var xSquared = Expression.Multiply(xParameter, xParameter);
+var ySquared = Expression.Multiply(yParameter, yParameter);
+var sum = Expression.Add(xSquared, ySquared);
+```
+
+Next, you need to create a method call expression for the call to
+`Math.Sqrt`.  This requires creating an expression that calls
+`Math.Sqrt`, and then updating that expression so that the
+arguments represent the sum of xSquared and ySquared:
+
+```cs
+Expression<Func<double, double>> sqrt = (x) => Math.Sqrt(x);
+var methodCall = sqrt.Body as MethodCallExpression;
+var distance = methodCall.Update(default(Expression), new List<Expression> { sum });
+```
+
+> Those familiar with the Reflection APIs may wonder why I didn't use
```

Comment:
I'm thinking it may be good to drop this aside.  I don't think too many people would *prefer* to use reflection here, and bringing up the notion that .NET Core doesn't have something that people may be interested in doesn't quite feel appropriate for the doc.